mlx_graphs/datasets/dataset.py

```diff
@@ -26,6 +31,7 @@ def __init__(
         self._name = name
         self._base_dir = base_dir
 
+        self.graphs: np.array[GraphData] = None
```

Comment:
Why is `self.graphs` a `np.array`? I don't see any issues in making it a `list[GraphData]`